### DISCUSSION TITLE
Tooltip - NextJS 13

### DISCUSSION THREAD

**Discussion by louislefebvre** [radix-ui/primitives#1782] [Help] [answered]
Hello :)

With NextJS 13 introducing their new app directory (currently in beta), as by default layouts and pages are SSR, I was wondering how [Radix Tooltip provider](https://www.radix-ui.com/docs/primitives/components/tooltip#provider) can be used globally over the app?

If for example, I wrap in my root layout body my childrens with <Tooltip.Provider> component, I get an error from Next saying that it has to be a client side component as it uses useEffect.

```ts
import "styles/globals.css";
import * as Tooltip from "@radix-ui/react-tooltip";

interface RootLayoutProps {
  children: React.ReactNode;
}

export default function RootLayout({ children }: RootLayoutProps) {
  return (
    <html lang="en">
      <head />
      <body className="body">
        <Tooltip.Provider>
          {children}
        </Tooltip.Provider>
      </body>
    </html>
  );
}
```

```
index.js?dd0a:598 Uncaught Error: 

You're importing a component that needs useEffect. It only works in a Client Component but none of its parents are marked with "use client", so they're Server Components by default.

   ,----
 4 | import { useLayoutEffect, useEffect } from 'react';
   :                           ^^^^^^^^^
   `----


Maybe one of these should be marked as a client entry with "use client":
  node_modules/.pnpm/@floating-ui+react-dom@0.7.2_2zx2umvpluuhvlq44va5bta2da/node_modules/@floating-ui/react-dom/dist/floating-ui.react-dom.mjs
  node_modules/.pnpm/@radix-ui+react-popper@1.0.1_2zx2umvpluuhvlq44va5bta2da/node_modules/@radix-ui/react-popper/dist/index.js
  node_modules/.pnpm/@radix-ui+react-tooltip@1.0.2_2zx2umvpluuhvlq44va5bta2da/node_modules/@radix-ui/react-tooltip/dist/index.js
  app/layout.tsx
```

Thanks a lot.

**Comment by andy-hook** [comment, marked as answer]
Hey @louislefebvre, we've yet to add out of the box support for the app directory, in the meantime I'd suggest following [the guidance provided in the official next docs](https://beta.nextjs.org/docs/rendering/server-and-client-components#rendering-third-party-context-providers-in-server-components) as relates to third party packages.

**Reply by louislefebvre** [reply]
Thank you very much for your reply!

**Reply by jkohlin** [reply]
…so [what it says in the docs](https://www.radix-ui.com/docs/primitives/overview/server-side-rendering#overview) is not really true then? 
> Radix Primitives can be rendered on the server

then maybe it should say …(if you wrap it in a "use client" directive, so that it doesn't render on the server…)

**Reply by andy-hook** [reply]
> so [what it says in the docs](https://www.radix-ui.com/docs/primitives/overview/server-side-rendering#overview) is not really true then?

SSR is not the same as RSC. That page is accurate with regards to SSR, though I agree some additional guidances specific to RSC would be helpful.

**Reply by jkohlin** [reply]
Ok, I see. So Radix is not a good match for my project then? I thought static-ish components like Avatar etc could be used in server components since they don't really (in my head) require client state. Do you know of any good component library that only rely on props and work in RSC's
